synapse/storage/state.py

```diff
@@ -89,6 +89,69 @@ def _get_current_state_ids_txn(txn):
             _get_current_state_ids_txn,
         )
 
+    # FIXME: how should this be cached?
```

Comment:
I don't understanding the caching stuff to know how best to handle this - or whether this method should be combined with get_current_state_ids above, and somehow share the same caching.

Reply:
I guess you've got three choices:

1. treat each call to `get_filtered_current_state_ids` as a separate cacheable thing - so for example if you do a lookup of `get_filtered_current_state_ids(room, None, None)`, a lookup of `get_filtered_current_state_ids(room, [(t, k)], None)` would not hit the cache. To do this, you just need an @cached decorator, plus invalidation calls.

2. Implement this in terms of a call to `get_current_state_ids` followed by a filter. This might increase cache hits, but cache misses will of course be very expensive.

3. use a DictionaryCache, like `_state_group_cache` uses, since `DictionaryCache` has support for partial caches. The keys would be room ids rather than state group ids. again you'd need to consider invalidation.

4. Not bother with caching and see how much it hurts?